features/0454-present-proof-v2/README.md

```diff
@@ -0,0 +1,289 @@
+
+Diagrams in this protocol were made in draw.io. To make changes:
+
+- upload the drawing HTML from this folder to the [draw.io](https://draw.io) site (Import From...GitHub),
+- make changes,
+- export the picture and HTML to your local copy of this repo, and
+- submit a pull request.
+
+### Roles
+
+The roles are `verifier` and `prover`.  The `verifier` requests the presentation of a proof and verifies the presentation, while the `prover` prepares the proof and presents it to the verifier. Optionally, although unlikely from a business sense, the `prover` may initiate an instance of the protocol using the `propose-presentation` message.
+
+### States
+
+#### States for Verifier
+
+* request-sent
+* proposal-received
+* presentation-received
+* abandoned
+* done
+
+#### States for Prover
+
+* request-received
+* proposal-sent
+* presentation-sent
+* abandoned
+* done
+
+For the most part, these states map onto the transitions shown in the choreography diagram in obvious ways. However, a few subtleties are worth highlighting:
+
+* The final transitions for the verifier (Send Presentation Reject, Send Presentation Ack, and Receive Presentation Reject) all result in a final `done` state, but this `done` may or may not be the verifiers intended outcome. We may want to tweak this in a future version of the protocol.
+
+> To Do: Should the previous sentence be addressed? Removed?
+
+* A similar situation exists with the final transitions for the Prover and its final transitions (Send Presentation Reject, Receive Presentation Ack, and Receive Presentation Reject).
+* When a Prover makes a (counter-)proposal, it transitions to the `proposal-sent` state. This state is only present by implication in the choreography diagram; it essentially equates to the null or begin state in that the Prover does nothing until a presentation request arrives, triggering the leftmost transition for the Prover.
+
+Errors might occur in various places. For example, a Verifier might time out waiting for the Prover to supply a presentation. Errors trigger a `problem-report`. In this version of the protocol, all errors cause the state of both parties (the sender and the receiver of the `problem-report`) to revert to `null` (meaning it is no longer engaged in the protocol at all). Future versions of the protocol may allow more granular choices.
+
+> To Do: Should the previous sentence be addressed in this version? Should it be removed?
+
+### Choreography Diagram
+
+![present proof](credential-presentation.png)
```

Comment:
The verifier's "Verify `presentation`" activity should be substituted for a decision point, eg. "Is `presentation` valid?" with Yes or No as outcomes.

Reply:
Is "Receive Presentation Reject" [here](https://github.com/hyperledger/aries-rfcs/pull/486/files#diff-6f5d41c9a81691c26389a4e0172b90a1R68) the "No" outcome for my suggestion above ?

Reply:
I think this may have been left as a TODO [here](https://github.com/hyperledger/aries-rfcs/pull/486#issuecomment-649129451)? @swcurran correct me if I'm wrong

Reply:
Based on the last Aries WG call (20200624) call, I think we want to add a paragraph on "Verify" (cryptographic) and an optional "Validate" (business rules) and suggest that the outcome from the Verifier be the outcome of the validation.

I think that is what you are saying as well in this comment, right @llorllale ?

I will put that into the RFC, and perhaps update the picture if I can figure out how.  Yes, I know, there are directions :-).

Reply:
> Based on the last Aries WG call (20200624) call, I think we want to add a paragraph on "Verify" (cryptographic) and an optional "Validate" (business rules) and suggest that the outcome from the Verifier be the outcome of the validation.

OK, but I'm not sure of making that distinction in the RFC...